src/main.c

```diff
@@ -157,6 +158,12 @@ int main(int argc, char **argv)
 	int index;
 	int ret;
 
+	/* Set locale to system locale
+	 * Change from the standard (C) to system locale, so libarchive can
+	 * handle filename conversions.
+	 * More information on the libarchive problem:
+	 * https://github.com/libarchive/libarchive/wiki/Filenames */
+	setlocale(LC_ALL, "");
```

Comment:
I would recommend using `LC_CTYPE` here instead of `LC_ALL` as libarchive should only care about the `LC_CTYPE` and setting `LC_ALL` could cause unexpected side effects, I've found on some embedded linux systems that `LC_ALL` doesn't work while `LC_CTYPE` does.